src/argilla_sdk/records/_io/_datasets.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from typing import Dict, List, Type, Union
+
+try:
+    from datasets import Dataset as HFDataset
+
+    datasets_imported = True
```

Comment:
(Just to keep in mind). So we should centralize the dataset usage in this module. If we want to use `datasets` in another module, we should use this one as an intermediate.